tests/test_config.py

```diff
@@ -19,25 +17,94 @@ def test_empty_config_file_is_created_if_not_present():
         assert config_file.read_text() == """[connections]\n"""
 
 
+@mock.patch.dict(os.environ, {}, clear=True)
+def test_get_connection_from_file(test_snowcli_config):
+    cm = CliConfigManager(file_path=test_snowcli_config)
+    cm.read_config()
+
+    assert cm.get_connection("full") == {
+        "account": "dev_account",
+        "user": "dev_user",
+        "host": "dev_host",
+        "port": 8000,
+        "protocol": "dev_protocol",
+        "role": "dev_role",
+        "schema": "dev_schema",
+        "database": "dev_database",
+        "warehouse": "dev_warehouse",
+    }
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "SNOWFLAKE_CONNECTIONS_DEV_DATABASE": "database_foo",
+        "SNOWFLAKE_CONNECTIONS_DEV_WAREHOUSE": "large",
+    },
+    clear=True,
+)
 def test_environment_variables_override_configuration_value(test_snowcli_config):
     cm = CliConfigManager(file_path=test_snowcli_config)
     cm.read_config()
 
-    assert cm.get("connections", "default", key="warehouse") == "xs"
-    with mock.patch.dict(
-        os.environ, {"SNOWFLAKE_CONNECTIONS_DEFAULT_WAREHOUSE": "foo42"}
-    ):
-        assert cm.get("connections", "default", key="warehouse") == "foo42"
+    assert cm.get_connection("dev") == {
+        "database": "database_foo",
+        "schema": "test_public",
+        "role": "test_role",
+        "warehouse": "large",
+    }
 
 
+@mock.patch.dict(
+    os.environ,
+    {
+        "SNOWFLAKE_CONNECTIONS_EMPTY_ACCOUNT": "some_account",
+        "SNOWFLAKE_CONNECTIONS_EMPTY_DATABASE": "test_database",
+        "SNOWFLAKE_CONNECTIONS_EMPTY_WAREHOUSE": "large",
+    },
+    clear=True,
+)
 def test_environment_variables_works_if_config_value_not_present(test_snowcli_config):
     cm = CliConfigManager(file_path=test_snowcli_config)
     cm.read_config()
 
-    with pytest.raises(Exception):
-        assert cm.get("connections", "2o3nei23ne2oixn", key="warehouse") == "xs"
+    assert cm.get_connection("empty") == {
```

Comment:
this test doesn't check what its name tells

Reply:
fixed